src/Annotation/Provider.php

```diff
@@ -14,8 +14,20 @@ final class Provider implements Annotation
 {
     /**
      * Optionnal prefix for provider fields.
-     *
-     * @var string
      */
     public string $prefix;
+
+    /**
+     * The default target types to attach the provider queries to.
+     *
+     * @var array<string>
+     */
+    public array $targetTypeQuery;
```

Comment:
Wouldn't it be better to name the property `$targetQueryTypes`?

Reply:
You are right, but I wanted to stay consistent with the `targetType` of the `@Query` and let people know that `targetTypeQuery` and `targetType` are about the same feature.
Also, now it supports multiple types, so `targetTypes` would be better but it wouldn't be BC.

Reply:
You can add all names and mark old ones deprecated, so that it stays BC. It just requires a little more logic to check both fields and a comment, e.g:
```PHP
// TODO: Remove old property check in 1.1
$targetTypes = $anno->targetTypes ?? $anno->targetType ?? null;
```

Reply:
So, I marked the old one as deprecated and handle both cases as you suggest (and left one old `targetType` in the tests for now).
Should I also update the `UPGRADE-1.0.md` ?

Reply:
I think it will be required in `UPGRADE-1.1.md` when the property is actually removed

Reply:
As always, thank you @murtukov  for the great review!